src/main/java/com/example/bookstore/service/shoppingcart/impl/ShoppingCartServiceImpl.java

```diff
@@ -0,0 +1,80 @@
+package com.example.bookstore.service.shoppingcart.impl;
+
+import com.example.bookstore.dto.cartitem.CartItemDto;
+import com.example.bookstore.dto.cartitem.CreateCartItemRequestDto;
+import com.example.bookstore.dto.cartitem.UpdateCartItemRequestDto;
+import com.example.bookstore.dto.shoppingcart.ShoppingCartDto;
+import com.example.bookstore.exception.EntityNotFoundException;
+import com.example.bookstore.mapper.CartItemMapper;
+import com.example.bookstore.mapper.ShoppingCartMapper;
+import com.example.bookstore.model.CartItem;
+import com.example.bookstore.model.ShoppingCart;
+import com.example.bookstore.model.User;
+import com.example.bookstore.repository.cartitem.CartItemRepository;
+import com.example.bookstore.repository.shoppingcart.ShoppingCartRepository;
+import com.example.bookstore.service.shoppingcart.ShoppingCartService;
+import jakarta.transaction.Transactional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+
+@RequiredArgsConstructor
+@Service
+public class ShoppingCartServiceImpl implements ShoppingCartService {
+    private final ShoppingCartRepository shoppingCartRepository;
+    private final CartItemRepository cartItemRepository;
+    private final ShoppingCartMapper shoppingCartMapper;
+    private final CartItemMapper cartItemMapper;
+
+    @Override
+    public ShoppingCartDto getShoppingCart(Long id) {
+        ShoppingCart shoppingCart = getShoppingCartByUserId(id);
+        return shoppingCartMapper.toDto(shoppingCart);
+    }
+
+    @Override
+    @Transactional
+    public ShoppingCartDto addBookToShoppingCart(
+            Long id, CreateCartItemRequestDto cartItemRequestDto) {
+        ShoppingCart shoppingCart = getShoppingCartByUserId(id);
```

Comment:
Find book in DB first
Check if book already exists in shop cart. If so -> increase quantity, if no -> add cart item